### ISSUE TITLE
Change to upstream `better-discord.js`

### ISSUE THREAD

**Issue by LecrisUT** [Half-Shot/matrix-appservice-discord#768]
Looking through Suronume's changes to the `better-discord.js`, I see that it simply adds `editMessages`, however this is already present in upstream
https://github.com/discordjs/discord.js/blob/b2dfb2ec70c33d30cf9b4a0b6757fbc6c750826c/packages/discord.js/src/structures/Webhook.js#L315
The upstream also includes threaded interface for future use, so how about refactoring this dependency? Api is different, but it shouldn't be hard to change.

**Comment by Sorunome** [comment]
sorus thing does way more, it also e.g. adds usertoken support

10. 1. 2022 8:36:49 LecrisUT ***@***.***>:

**Comment by LecrisUT** [comment]
Do you have a list of changes or PRs so we can check if they are implemented upstream? I see that there are differences in implementation so simply merging upstream will be difficult.

**Comment by Sorunome** [comment]
nope, but the big things are usertoken and friends management. Also, plop into the mx-puppet-bridge, daniel is also tackeling the aame issue there.

10. 1. 2022 8:44:57 LecrisUT ***@***.***>:

**Comment by LecrisUT** [comment]
Kay, thanks for the info.

**Comment by erkinalp** [comment]
> it also e.g. adds usertoken support

What is bad with that?

**Comment by dsonck92** [comment]
If logging in as regular users is out of scope for this bridge (which I would suggest it is), then I second the decision to go to upstream again. In my eyes, the appservice series of bridges are best off focusing on bridging in a bot/webhook way, intended for "other chat"-admins to invite to their equivalent of rooms. Essentially the way some IRC<->Discord bridges are made. While bridges dedicated to user bridging through puppeting should focus on doing that, by any means necessary (user tokens, talking client API-like, etc.)

Essentially I see them as "official" and "personal" bridges. The "official" bridges being targeted to moderators/admins of projects that happen to have multiple chats (e.g. software projects on Slack/Discord). The "personal" bridges being targeted to users that happen to require access to Discord/Slack/etc. because some of their peers, or products (e.g. games, or software projects on Slack/Discord) haven't seen Matrix yet (or refuse to use it).

But that's just my view on it. If this project feels it should implement puppeting on top of bot usage, then there's little possibility to change this as it requires the fork.

**Comment by erkinalp** [comment]
@dsonck92 If you want to support both, you can use fosscord/fosscord.js. It is a fork of discord.js that allows changing the endpoint and use with either user or bot access tokens.

**Comment by dsonck92** [comment]
Oh I don't maintain either project anymore (for a puppeting bridge I just lift on mautrix-discord which isn't node.js based), and this project I just use for bridging my own server over to Matrix as I have the power to do so. But fosscord/fosscordjs sounds like a good idea then.
